packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -55,49 +36,69 @@ export const useMagentoRoute = props => {
         };
     }, []);
 
-    // If the entry for this pathname is a redirect, perform the redirect.
+    const storeCodeResult = useQuery(getStoreCode, {
+        fetchPolicy: 'cache-and-network',
+        nextFetchPolicy: 'cache-first'
+    });
+
+    const { data: storeCodeData } = storeCodeResult;
+    const store = storeCodeData && storeCodeData.storeConfig.code;
+    const routeKey = getRouteKey(pathname, store);
+    const routeData = responses.get(routeKey);
+
+    const urlResult = useQuery(RESOLVE_URL, {
+        fetchPolicy: 'cache-and-network',
+        nextFetchPolicy: 'cache-first',
+        variables: { url: pathname }
```

Comment:
Will this work when using store code in url? You may have a pathname like `/fr/shop-the-look.html` in this case.

Or I could be wrong as we should be setting the base properly when updating store code, so maybe it removes the prefix.

Reply:
Yes; it should, since the store code is a header and we apply with Apollo Link.

Reply:
Confirmed. This works because we set the `basename` on the router, so `pathname` would be `/shop-the-look.html` in that case.

<img width="1719" alt="PDP" src="https://user-images.githubusercontent.com/2110986/100002375-b673b700-2d81-11eb-9aba-5fd6225c50e6.png">